.github/workflows/shared-publish-to-docker-versioned.yaml

```diff
@@ -0,0 +1,116 @@
+name: Shared Pipeline to build and publish images to Docker
+on:
+  workflow_call:
+    inputs:
+      release_type:
+        description: The type of version number to return. Must be one of [Snapshot, Patch, Minor or Major]
+        required: true
+        type: string
+      publish_vulnerabilities:
+        description: If true, will attempt to publish any vulnerabilities to GitHub. Defaults to true. Set to false for private repos.
+        type: string
+        default: true
+      version_number_input:
+        description: The version number to publish for the docker repo
+        type: string
+        default: ''
+      force_release:
+        description: If 'yes', will force the creation a release, if 'no' will not create a release. 'branch' will use release_type and the branch to determine if a release should be created.
+        type: string
+        default: 'branch'
+      docker_file:
+        description: The Dockerfile used to build and publish Docker image
+        type: string
+        default: ''
+      docker_image_name:
+        description: The docker image name
+        required: true
+        type: string
+      docker_registry:
+        description: The docker registry
+        required: true
+        type: string
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+jobs:
+  buildImage:
+    name: Build Image
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      security-events: write
+      packages: write
+      pull-requests: write
+    outputs:
+      jar_version: ${{ inputs.version_number_input }}
+      image_tag: ${{ steps.updatePom.outputs.image_tag }}
+    steps:
+      - name: Setup
+        id: setup
+        uses: IABTechLab/uid2-shared-actions/actions/shared_publish_setup@kcc-UID2-2674-implement-shared-publish-to-docker-versioned
+        with:
+          release_type: ${{ inputs.release_type }}
+      
+      - name: Checkout uid2-shared-actions repo
+        uses: actions/checkout@v3
+        with:
+          ref: kcc-UID2-2674-implement-shared-publish-to-docker-versioned
+          repository: IABTechLab/uid2-shared-actions
+          path: uid2-shared-actions
+
+      - name: Set version number
+        id: version
+        uses: IABTechLab/uid2-shared-actions/actions/version_number@main
+        with:
+          type: ${{ inputs.release_type }}
+          version_number: ${{ inputs.version_number_input }}
+          branch_name: ${{ github.ref }}
+
+      - name: Update package.json
```

Comment:
I think we should be able to find or write an action to update the json files without having to clone the shared-actions repo. Feels wrong to me to clone the whole repo just to execute a script. 
You could do the same as the version_number action does, rather than calling this script